### PR TITLE
Include the git commit id in the version string

### LINKED ISSUE #503: Include git commit version in the about window
I would suggest adding some kind of info like what git revision (possibly use `git describe` which currently outputs `v0.4-223-ge330d27d`) is being used, so that when an development version is installed from git, you can see what commit it is. This would make it easier to determine what development version you're on and how far you're behind.

Possibly the `GTG/gtg.in` file could be used to set the variable, since it is processed anyway (`.in`).

I'd suggest adding this to the about window, since the version is listed there, and the `--version` command.

### PR DESCRIPTION
**Pull request by frainz-de**
This is my attempt to fix #503 .

Use the output of "git describe --dirty" as the version string.
The info is filled in by meson.
I'm not sure if I'm happy about it because the version is now always taken from git describe.
Maybe the commit id should be somewhere else?

### CODE REVIEW

File: GTG/core/meson.build
Comment (Neui) [inline GTG/core/meson.build:2]: I think it should handle the case when `git` isn't available or fails (such as when the source tarball or the download-as-zip on github is being used). (Actually didn't test what happens)
Maybe fall back to the version as [defined in the main `meson.build` file](https://github.com/getting-things-gnome/gtg/blob/d7bbcd5b2a9d7d54b2b28278e5bbf6fd264b302b/meson.build#L2)? (Seems `meson.project_version()` would return that version).

File: GTG/core/meson.build
Comment (Neui) [inline GTG/core/meson.build:12]: I think you should also use the `/` operator here, just in case: `infodir / 'GTG' / 'core'`

File: GTG/core/info.py.in
Comment (Neui) [inline GTG/core/info.py.in:39]: When a release gets made, it'll be `VERSION = 'v0.5'`, which diverges from the "expected" format.
Maybe save it in a new variable and use some python code here to determine what to use?
This could also be fine, but I'd rather point it out.

**Review by Neui — CHANGES_REQUESTED** [review]

File: GTG/core/info.py.in
Comment (frainz-de) [reply GTG/core/info.py.in:39]: I'm not sure what you mean with that. Does the version format change in the next version?

File: GTG/core/meson.build
Comment (frainz-de) [reply GTG/core/meson.build]: It's a bit longish now, but it works both when git is not present and when we are not building from a repo

File: GTG/core/info.py.in
Comment (Neui) [reply GTG/core/info.py.in:39]: Well, `git describe` uses (annotated) tags as a base, and when on a annotated tag, it directly uses the tag name. However, the format of those are `vX.Y`, see: https://github.com/getting-things-gnome/gtg/tags , which as you can see differs from the format uses here (`X.Y.Z`). If the maintainers/others are OK with that, tehn I think it could stay as-is. I haven't checked how it looks like on the GUI.

Comment (frainz-de) [reply GTG/core/info.py.in:39]: It looks like this now. Do you know where it gets the branch name from? I couldn't find any other use of git in the source.

Screenshot from 2020-11-19 21-13-14

Comment (Neui) [reply GTG/core/info.py.in:39]: It is the title and the "branch" is actually the folder name here: https://github.com/getting-things-gnome/gtg/blob/d7bbcd5b2a9d7d54b2b28278e5bbf6fd264b302b/scripts/debug.sh#L48 and https://github.com/getting-things-gnome/gtg/blob/master/GTG/gtg.in#L102

I also looked a bit and it seems `VERSION` is only used twice, for displaying not parsing, so this would be fine I think. I'll however want others feedback, whenever it would be OK to change the "format". I guess I might also be too picky here.

**Comment by frainz-de** [comment]
What's the status here? Are we waiting for the maintainers to comment or is there something left to fix before merging?

**Comment by diegogangl** [comment]
Hi @frainz-de, I've been going through some family health issues so I haven't been able to spend time with gtg recently. This PR LGTM, so merging. Thanks! (also thanks to @Neui as usual for the help)

**Pull request merged**
